server/src/test/java/org/elasticsearch/index/mapper/RangeFieldMapperTests.java

```diff
@@ -261,101 +176,51 @@ protected void doTestDecimalCoerce(String type) throws IOException {
-        IndexableField pointField = fields[1];
-        assertEquals(2, pointField.fieldType().pointIndexDimensionCount());
-        assertFalse(pointField.fieldType().stored());
         IndexableField storedField = fields[2];
-        assertTrue(storedField.fieldType().stored());
-        String expected = type.equals("ip_range") ? InetAddresses.toAddrString((InetAddress) getMax(type)) : getMax(type) + "";
-        assertThat(storedField.stringValue(), containsString(expected));
+        return storedField.stringValue();
     }
 
-    public void testIllegalArguments() throws Exception {
-        Exception e = expectThrows(
-            MapperParsingException.class,
-            () -> createDocumentMapper(fieldMapping(b -> b.field("type", RangeType.INTEGER.name).field("format", DATE_FORMAT)))
-        );
-        assertThat(e.getMessage(), containsString("should not define a dateTimeFormatter"));
-    }
+    public final void testNullBounds() throws IOException {
 
-    public void testSerializeDefaults() throws Exception {
-        for (String type : types()) {
-            DocumentMapper docMapper = createDocumentMapper(fieldMapping(b -> b.field("type", type)));
-            RangeFieldMapper mapper = (RangeFieldMapper) docMapper.mapping().getRoot().getMapper("field");
-            XContentBuilder builder = XContentFactory.jsonBuilder().startObject();
-            ToXContent.MapParams params = new ToXContent.MapParams(Collections.singletonMap("include_defaults", "true"));
-            mapper.doXContentBody(builder, params);
-            String got = Strings.toString(builder.endObject());
-
-            // if type is date_range we check that the mapper contains the default format and locale
-            // otherwise it should not contain a locale or format
-            assertTrue(got, got.contains("\"format\":\"strict_date_optional_time||epoch_millis\"") == type.equals("date_range"));
-            assertTrue(got, got.contains("\"locale\":" + "\"" + Locale.ROOT + "\"") == type.equals("date_range"));
-        }
-    }
+        // null, null => min, max
+        assertNullBounds(b -> b.startObject("field").nullField("gte").nullField("lte").endObject(), true, true);
 
-    public void testIllegalFormatField() throws Exception {
-        Exception e = expectThrows(
-            MapperParsingException.class,
-            () -> createMapperService(fieldMapping(b -> b.field("type", "date_range").array("format", "test_format")))
-        );
-        assertThat(e.getMessage(), containsString("Invalid format: [[test_format]]: Unknown pattern letter: t"));
+        // null, val => min, val
+        Object val = rangeValue();
+        assertNullBounds(b -> b.startObject("field").nullField("gte").field("lte", val).endObject(), true, false);
+
+        // val, null -> val, max
+        assertNullBounds(b -> b.startObject("field").field("gte", val).nullField("lte").endObject(), false, true);
     }
 
     @Override
```

Comment:
Not technically part of your change here, but I notice the TODO below is dependent on closure of an issue that is now closed?

Reply:
The issue is closed but the problem remains, I think.